widgets/deploymentsView/src/columns.tsx

```diff
@@ -86,30 +114,54 @@ const partialDeploymentsViewColumnDefinitions: Record<
     subenvironmentsCount: {
         label: <Stage.Basic.Icon name="object group" />,
         width: '1em',
-        render() {
-            // TODO(RD-1742): display correct number of subenvironments and their status
-            return '0';
+        // NOTE: properties come from the API. They are not prop-types (false-positive)
+        /* eslint-disable camelcase, react/prop-types */
+        // TODO(RD-1839): remove default values
+        render({ sub_environments_count = 0, sub_environments_status = SubdeploymentStatus.Good }) {
+            const iconName = subdeploymentStatusToIconMapping[sub_environments_status];
+            const icon = iconName && renderStatusIcon(iconName);
+
+            return (
+                <div className="subdeploymentColumn">
+                    {sub_environments_count} {icon}
+                </div>
+            );
         }
     },
     subservicesCount: {
         label: <Stage.Basic.Icon name="cube" />,
         width: '1em',
-        render() {
-            // TODO(RD-1742): display correct number of subservices and their status
-            return '0';
+        // TODO(RD-1839): remove default values
+        render({ sub_services_count = 0, sub_services_status = SubdeploymentStatus.Good }) {
+            const iconName = subdeploymentStatusToIconMapping[sub_services_status];
+            const icon = iconName && renderStatusIcon(iconName);
+
+            return (
+                <div className="subdeploymentColumn">
+                    {sub_services_count} {icon}
+                </div>
+            );
         }
+        /* eslint-enable camelcase, react/prop-types */
     }
 };
 
+const subdeploymentStatusToIconMapping: Record<SubdeploymentStatus, StatusIconName | undefined> = {
+    [SubdeploymentStatus.InProgress]: 'inProgress',
+    [SubdeploymentStatus.Good]: undefined,
+    [SubdeploymentStatus.Failed]: 'requiresAttention',
+    [SubdeploymentStatus.Pending]: undefined
```

Comment:
If you simplify the code (see my first comment), you should be able to avoid using strings and have something like:
```suggestion
    [SubdeploymentStatus.InProgress]: statusIcon.inProgress,
    [SubdeploymentStatus.Good]: undefined,
    [SubdeploymentStatus.Failed]: statusIcon.requiresAttention,
    [SubdeploymentStatus.Pending]: undefined
```

Reply:
Not relevant anymore. Resolving.